### PR TITLE
Add 3 more Helmholtz-Ellis accidentals

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
the last missing ones, just to make the set complete

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Interesting, we were too close to the max. number of initializers for an `std::array` to add 3 more entries, and on all 3 platforms!

Any idea anyone how to get around this?

**Comment by cbjeukendrup** [comment]
It was just because we specify `size_t(SymId::lastSym) + 1` as the number of items in the array :)

File: src/engraving/types/symnames.cpp [hunk 1]
```diff
@@ -3453,9 +3453,9 @@ constexpr const std::array<const char*, size_t(SymId::lastSym) + 1> SymNames::s_
     QT_TRANSLATE_NOOP("engraving/sym", "Double sharp raised by three syntonic commas"),
     QT_TRANSLATE_NOOP("engraving/sym", "Double sharp lowered by two syntonic commas"),
     QT_TRANSLATE_NOOP("engraving/sym", "Double sharp raised by two syntonic commas"),
-    "Enharmonically reinterpret accidental almost equal to",
-    "Enharmonically reinterpret accidental equals",
-    "Enharmonically reinterpret accidental tilde",
+    QT_TRANSLATE_NOOP("engraving/sym", "Enharmonically reinterpret accidental tilde"),
+    QT_TRANSLATE_NOOP("engraving/sym", "Enharmonically reinterpret accidental almost equal to"),
+    QT_TRANSLATE_NOOP("engraving/sym", "Enharmonically reinterpret accidental equals"),
```
Comment (cbjeukendrup) [inline src/engraving/types/symnames.cpp:3458]: I believe the order of these must match that in `symid_p.h`

Comment (Jojo-Schmitz) [reply src/engraving/types/symnames.cpp:3458]: Doesn't it?

Comment (Jojo-Schmitz) [reply src/engraving/types/symnames.cpp]: You're of course right

**Comment by Jojo-Schmitz** [comment]
> It was just because we specify `size_t(SymId::lastSym) + 1` as the number of items in the array :)

Seems to be one too much as far as I can tell?

**Comment by cbjeukendrup** [comment]
If you count `SymId::lastSym` too, there are `SymId::lastSym + 1` SymIds, because the enum cases start counting from zero (just like the list `{0, 1, 2}` contains `2+1=3` elements). 
It's debatable whether the list should contain an entry for `SymId::lastSym`, but I guess someone did that on purpose to prevent a read-outside-bounds when accidentally passing `SymId::lastSym` to one of those methods to retrieve symbol names.

**Comment by Jojo-Schmitz** [comment]
Yes, but there is no such symbol, just that enum, no element at that position in those `std::array`s

**Comment by cbjeukendrup** [comment]
That's true, but that doesn't really harm; then those array entries are simply filled with zeros, which still fulfils the purpose of not causing read-outside-bounds.

File: src/engraving/dom/types.h [hunk 2]
```diff
@@ -249,9 +249,9 @@ enum class AccidentalType {
     SHARP_31,
     FLAT_53,
     SHARP_53,
-    //EQUALS_ALMOST,
-    //EQUALS,
-    //TILDE,
+    EQUALS_ALMOST,
+    EQUALS,
+    TILDE,
```
Comment (cbjeukendrup) [inline src/engraving/dom/types.h:254]: Do you by chance still remember why these were commented out initially?

**Review by cbjeukendrup — APPROVED** [review]

File: src/engraving/dom/types.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/dom/types.h:254]: No, I don't recall

**Comment by Jojo-Schmitz** [comment]
Yes, better one too large than one too small ;-)

File: src/engraving/dom/types.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/dom/types.h:254]: Probably because I didn't know which cent offsets to use

**Comment by zacjansheski** [comment]
Adding the new sharp to palette crashes MuseScore
[AddSharpCrash.zip](https://github.com/musescore/MuseScore/files/13890804/AddSharpCrash.zip)



https://github.com/musescore/MuseScore/assets/69917893/09e02c58-3499-4e71-83b1-388fb456aa51

**Comment by Jojo-Schmitz** [comment]
I can't reproduce, no crash, I can't add that sharp to the palette at all (but can add the ones left and right of it).

**Comment by zacjansheski** [comment]
Thank you!

It only happens on my MacOS Sonoma device

@cbjeukendrup would you kindly look at the diagnostic files and let me know if there is any cause for concern here?

[SonomaCrash.zip](https://github.com/musescore/MuseScore/files/13906229/SonomaCrash.zip)

**Comment by Jojo-Schmitz** [comment]
I checked your https://github.com/musescore/MuseScore/pull/20566 yesterday and couldn't find anything usefull

**Comment by cbjeukendrup** [comment]
In a local debug build, adding that sharp to the palette just doesn't work for me, like for @Jojo-Schmitz; in the build from GitHub, I get a crash, like @zacjansheski. The only thing I can say is that the crash has something to do with tooltips, but a few levels deep inside Qt, which means that it's either a Qt bug or the QML code contains something bad. This is the stack trace:
Scherm­afbeelding 2024-01-11 om 23 25 24
I can't really get any more info, which makes this situation somewhat difficult. 

If I delete the normal sharp from the palette first, then adding the "weird" sharp does work without problems. Perhaps that means that something still gets confused by the fact that those accidentals have the same symid... maybe because that results in `Accidental::subtypeUserName()` and more importantly `propertyToString(Pid::ACCIDENTAL_TYPE, …)` to be the same for these accidentals?

@Jojo-Schmitz Hopefully that gives some inspiration for how to debug this.

**Comment by sammik** [comment]
But there is no #4 Turkish accidental in SMuFL, is it?

https://w3c.github.io/smufl/latest/tables/turkish-folk-music-accidentals.html

**Comment by cbjeukendrup** [comment]
No, so therefore that uses the normal sharp symbol instead, but that causes problems because now the same symId is used for multiple kinds of accidentals. 

Perhaps there is a reason though that SMuFL doesn't contain a symbol for it; perhaps a 4-comma sharp simply doesn't exist. I can't imagine they left it out by accident... it would be great if someone with knowledge about Turkish music could confirm that.

**Comment by Jojo-Schmitz** [comment]
There's none, they do use the regular sharp instead. But with an 88.8 cent offset, so 11.1 less sharp than the regular one.

**Comment by Jojo-Schmitz** [comment]
That cent offset doesn't seem to work on that sharp, so maybe I better remove it from this PR?
Alternative would be to 'invent' a 4-comma sharp?

**Comment by zacjansheski** [comment]
Thank you Jojo!

I think it would be best to drop the sharp from this PR and move forward with the other 3.

**Comment by Jojo-Schmitz** [comment]
Done, along with a rebase

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved

**Pull request merged**